### PR TITLE
Unscope order shipping method

### PR DESCRIPTION
**Pull request by mbj**
* Completed orders with shipping methods deleted should still be valid

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb This is ready for review.

File: core/app/models/spree/order.rb [hunk 1]
```diff
@@ -139,6 +139,15 @@ def self.incomplete
...
     end
 
+    # The shipping method (even if deleted)
+    #
+    # @return [ShippingMethod]
+    #
+    # @api private
+    def shipping_method
+      ShippingMethod.unscoped { super() }
+    end
```
Comment (dkubb) [inline core/app/models/spree/order.rb]: You can get the same behaviour by scoping the association, eg:

```diff
diff --git a/core/app/models/spree/order.rb b/core/app/models/spree/order.rb
index 02aeeff..674e0e7 100644
--- a/core/app/models/spree/order.rb
+++ b/core/app/models/spree/order.rb
@@ -79,8 +79,8 @@ module Spree
     alias_attribute :shipping_address, :ship_address
     alias_attribute :ship_total, :shipment_total
 
-    belongs_to :store,           class_name: 'Spree::Store'
-    belongs_to :shipping_method, class_name: 'Spree::ShippingMethod'
+    belongs_to :store,                                class_name: 'Spree::Store'
+    belongs_to :shipping_method, -> { with_deleted }, class_name: 'Spree::ShippingMethod'
 
     has_many :line_items, -> { order("#{LineItem.table_name}.created_at ASC") }, dependent: :destroy, inverse_of: :order
     has_many :payments, dependent: :destroy
@@ -139,15 +139,6 @@ module Spree
       where(completed_at: nil).order(:created_at).reverse_order
     end
 
-    # The shipping method (even if deleted)
-    #
-    # @return [ShippingMethod]
-    #
-    # @api private
-    def shipping_method
-      ShippingMethod.unscoped { super() }
-    end
-
     # Return adjustments memory scope
     #
     # @return [Adjustment::Scopes]
```

Comment (mbj) [inline core/app/models/spree/order.rb]: The associated tickets in `paranoia` tell me this only works with rails 4.2, not on 4.1.

Comment (mbj) [inline core/app/models/spree/order.rb]: Reference: https://github.com/rubysherpas/paranoia/issues/109

**Comment by mbj** [comment]
@dkubb ready for the next pass.

File: core/app/models/spree/order.rb (hunk 1 above)
Comment (dkubb) [inline core/app/models/spree/order.rb]: > The associated tickets in paranoia tell me this only works with rails 4.2, not on 4.1.

We already use this in `Spree::LineItem` for the `#variant` and `#product` associations.

Also I made the change in the diff above and the specs continued to pass.

**Comment by mbj** [comment]
@dkubb Ready for the next pass.

**Pull request merged**